internal/jobcontainers/path.go

```diff
@@ -0,0 +1,177 @@
+package jobcontainers
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/Microsoft/hcsshim/internal/winapi"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/windows"
+)
+
+// This file emulates the path resolution logic that the Cexecsvc normally handles for regular
```

Comment:
I don't think it's very helpful for most readers to refer to CExecSvc here, just say it matches the normal Windows container launching behavior.

Reply:
I think this could be expanded a lot more. As an example:

```
searchPathForExe resolves a given command line string and returns the path to the executable that should be launched. The resolution logic may appear overcomplicated but is designed to match the logic used by standard Windows containers, as well as that used by CreateProcess (see notes for the lpApplicationName parameter).

The logic follows this set of steps:
- Step 1
- Step 2
- Step 3

Examples:
- Input: "C:\Program Files\sub dir\program name"
  Search order:
  - C:\Program.exe
  - C:\Program Files\sub.exe
  - C:\Program Files\sub dir\program.exe
  - C:\Program Files\sub dir\program name.exe
- Input: some other example
  Search order:
  - etc
  - etc

CreateProcess documentation: https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessa
```

Reply:
Good point, this was mainly a dev focused comment

Reply:
Changed

Reply:
Why does github not pop up new comments automatically -_-. Changing to something like what you've posted

Reply:
Put the big comment above getApplicationName as that's where all the logic is